.github/workflows/format.yml

```diff
@@ -54,7 +54,7 @@ jobs:
 
       - name: Run clang-tidy compilation
         run: |
-          cmake -BBuild -DENABLE_CLANG_TIDY=ON -DCLANG_TIDY_BINARY=clang-tidy-12 -DBUILD_EXAMPLES=ON -DBUILD_TESTS=on -DENABLE_WARNINGS=on .
+          CXX=g++-10 cmake -BBuild -DENABLE_CLANG_TIDY=ON -DCLANG_TIDY_BINARY=clang-tidy-12 -DBUILD_EXAMPLES=ON -DBUILD_TESTS=on -DENABLE_WARNINGS=on .
```

Comment:
If this is necessary, after the above, maybe we should do this as `-DCMAKE_CXX_COMPILER=g++-10`

Reply:
Updated workflows to use `DCMAKE_CXX_COMPILER` instead of the `CXX` env variable.